internal/services/netapp/netapp_volume_resource.go

```diff
@@ -553,6 +577,26 @@ func resourceNetAppVolumeUpdate(d *pluginsdk.ResourceData, meta interface{}) err
 		update.Properties.ThroughputMibps = utils.Float(throughputMibps.(float64))
 	}
 
+	if d.HasChange("smb_non_browsable_enabled") {
+		shouldUpdate = true
+		smbNonBrowsable := volumes.SmbNonBrowsableDisabled
+		update.Properties.SmbNonBrowsable = &smbNonBrowsable
+		if d.Get("smb_non_browsable_enabled").(bool) {
+			smbNonBrowsable := volumes.SmbNonBrowsableEnabled
+			update.Properties.SmbNonBrowsable = &smbNonBrowsable
+		}
```

Comment:
We can simplify this to make it more clear as:

```
		shouldUpdate = true
		if d.Get("smb_non_browsable_enabled").(bool) {
			update.Properties.SmbNonBrowsable = &volumes.SmbNonBrowsableEnabled
		} else {
            update.Properties.SmbNonBrowsable = &volumes.SmbNonBrowsableDisabled
        }
```

Reply:
@paulomarquesc This would produce an invalid operation error:
`invalid operation: cannot take address of volumes.SmbNonBrowsableEnabled (constant "Enabled" of type "github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2023-05-01/volumes".SmbNonBrowsable)`
It seems that the memory has to be allocated and then the address referenced. It seems using the address of a constant in go in this manner is not allowed